packages/kbn-interpreter/public/browser_registries.js

```diff
@@ -20,51 +20,64 @@
 import chrome from 'ui/chrome';
 import $script from 'scriptjs';
 
-let resolve = null;
+let resolvePromise = null;
 let called = false;
 
 let populatePromise = new Promise(_resolve => {
-  resolve = _resolve;
+  resolvePromise = _resolve;
 });
 
 export const getBrowserRegistries = () => {
   return populatePromise;
 };
 
+const loadBrowserRegistries = (registries) => {
+  const remainingTypes = Object.keys(registries);
+  const populatedTypes = {};
+
+  return new Promise(resolve => {
+    function loadType() {
+      if (!remainingTypes.length) {
+        resolve(populatedTypes);
+        return;
+      }
+      const type = remainingTypes.pop();
+      window.canvas = window.canvas || {};
+      window.canvas.register = d => registries[type].register(d);
+
+      // Load plugins one at a time because each needs a different loader function
+      // $script will only load each of these once, we so can call this as many times as we need?
+      const pluginPath = chrome.addBasePath(`/api/canvas/plugins?type=${type}`);
+      $script(pluginPath, () => {
```

Comment:
nit: Not sure, but can't we use something like `$script([plugin_path1, pluginPath2,...])` instead of doing this recursive stack?

Reply:
No, because the register function changes for each type.